File: src/main/java/com/cloudpi/cloudpi_backend/files_info/entities/FileEntity.java
package com.cloudpi.cloudpi_backend.files_info.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import java.util.Date;

@Getter
@Setter
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor

@Entity
@DiscriminatorValue("FILE")
public class FileEntity extends FilesystemObjectEntity {
    @Column(nullable = false)
    Long size;
    @Column(nullable = false)
    Date modifiedAt;
}

File: src/main/java/com/cloudpi/cloudpi_backend/files_info/entities/kotlin/FileEntity.kt
package com.cloudpi.cloudpi_backend.files_info.entities.kotlin

import com.cloudpi.cloudpi_backend.user.entities.UserEntity
import java.util.*
import javax.persistence.Column
import javax.persistence.DiscriminatorValue
import javax.persistence.Entity

//@Entity
//@DiscriminatorValue("FILE")
class FileEntity(
    id: Long,
    owner: UserEntity,
    permissions: List<FilePermissionEntity>,
    name: String,
    relativePath: String,
    parent: FilesystemIdEntity,
    root: RootDirectoryEntity,
    createdAt: Date,
    @Column(nullable = false)
    var size: Long,
    @Column(nullable = false)
    var modifiedAt: Date,


    ) : FilesystemObjectEntity(id, owner, permissions, name, relativePath, parent, root, createdAt)